checker.c

```diff
@@ -1,21 +1,48 @@
 #include <stdio.h>
 #include <assert.h>
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
+void checkbattery_Temp(float temperature)
+{
+  if(temperature <0 || temperature > 45)
+  {
+    printf("Temperature out of range!\n"); 
+  }
+  else
+  {
+     printf("Temperature is in normal range!\n"); 
+  }
+}
+
+void checkbattery_soc(float soc)
+{
+  if(soc < 20 || soc > 80)
```

Comment:
This function is very similar to the previous one. If the range comparison changes, you need to change in multiple places. Also, additional parameters will need more such repeats
Avoid this by making an abstraction to check ranges and write asserts for it